src/pam/error.rs

```diff
@@ -203,18 +201,11 @@ impl fmt::Display for PamError {
         match self {
             PamError::UnexpectedNulByte(_) => write!(f, "Unexpected nul byte in input"),
             PamError::Utf8Error(_) => write!(f, "Could not read input data as UTF-8 string"),
-            PamError::InvalidState => {
-                write!(
-                    f,
-                    "Could not initiate pam because the state is not complete"
-                )
-            }
-            PamError::Pam(tp, msg) => write!(f, "PAM returned an error ({tp:?}): {msg}"),
+            PamError::Pam(tp) => write!(f, "PAM returned an error ({tp:?}): {}", tp.get_err_msg()),
```

Comment:
I think I'd prefer actually seeing the `tp.get_err_msg` only and not the `{tp:?}`. This is stuff end-users might see and they shouldn't need to care about what some internal enum variant is called (see #981)

Reply:
Do the current error messages look better?